src/SimpleAzure.Storage.HybridQueue/HybridQueue.cs

```diff
@@ -105,21 +91,15 @@ public async Task AddMessageAsync<T>(T item, TimeSpan? initialVisibilityDelay, C
         }
 
         // Lets batch up these messages to make sure the awaiting of all the tasks doesn't go too crazy.
-        var contentsSize = contents.Count();
-        var finalBatchSize = contentsSize > batchSize
-                                 ? batchSize
-                                 : contentsSize;
-
-        foreach (var batch in contents.Chunk(finalBatchSize))
+        foreach (var batch in contents.Chunk(batchSize))

```

Comment:
The perf difference is because of the removal of the .Count() method, which is evil and should be avoided.

The logic change is entirely a guess based on reading the doc comments of Chujk, which day nothing about throwing if batch size is larger than the collection size